test/topology_experimental_raft/test_tombstone_gc.py

```diff
@@ -38,9 +41,104 @@ async def test_default_tombstone_gc(manager: ManagerClient, rf: int, tablets: bo
 @pytest.mark.parametrize("tablets", [True, False])
 async def test_default_tombstone_gc_does_not_override(manager: ManagerClient, rf: int, tablets: bool):
     _ = [await manager.server_add() for _ in range(2)]
-    cql = manager.cql
+    cql = manager.get_cql()
     tablets_enabled = "true" if tablets else "false"
     async with new_test_keyspace(cql, f"with replication = {{ 'class': 'NetworkTopologyStrategy', 'replication_factor': {rf}}} and tablets = {{ 'enabled': {tablets_enabled} }}") as keyspace:
         async with new_test_table(cql, keyspace, "p int primary key, x int", " with tombstone_gc = {'mode': 'disabled'}") as table:
             await cql.run_async(f"ALTER TABLE {table} add y int")
             check_tombstone_gc_mode(cql, table, "disabled")
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(reason="issue #15607")
+async def test_group0_tombstone_gc(manager: ManagerClient):
+    """
+    Regression test for #15607.
+
+    Test that the tombstones are being cleaned after the group 0 catching up.
+
+    Test #1:
+
+    Arrange:
+        - start 3 nodes
+    Act:
+        - create new group0 tombstones by updating the schema (create/alter/delete randome tables)
+    Assert:
+        - the tombstones are cleaned up eventually
+
+    Test #2:
+
+    Arrange:
+        - stop one of the nodes
+    Act:
+        - create new group0 tombstones by updating the schema (create/alter/delete randome tables)
+    Assert:
+        - the tombstones are not cleaned up (one of the nodes is down, so they can't catch up)
+
+    Test #3:
+
+    Arrange:
+        - start the node again
+    Act:
+        - tombstones exist from the previous test
+    Assert:
+        - the tombstones are cleaned up eventually
+    """
+    cfg = {
+        'group0_tombstone_gc_interval_in_ms': 1000,  # this is 1 hour by default
+    }
+    servers = [await manager.server_add(config=cfg) for _ in range(3)]
+    logs = [await manager.server_open_log(srv.server_id) for srv in servers]
+    marks = [await log.mark() for log in logs]
+
+    cql = manager.get_cql()
+    hosts = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+
+    # create/alter/drop a few tables
+    async def alter_system_schema():
+        for _ in range(5):
+            async with new_test_table(cql, keyspace, "p int primary key, x int") as table:
+                await cql.run_async(f"ALTER TABLE {table} add y int")
+
+    def has_tombstone(row):
+        if row.metadata is None:
+            return False
+        metadata = json.loads(row.metadata)
+        tombstone = metadata.get("tombstone")
+        return tombstone is not None
+
+    async def tombstone_gc_completed():
+        count_rows_tombstone_total = 0
+        for tbl in ("raft", "scylla_local"):
```

Comment:
`system.raft` is not a group 0 table, it's a 100% local table.

Since you're performing schema changes, why not pick some table in `system_schema` for testing?

Reply:
done